NineMensMorris/src/singleplayergame.cpp

```diff
@@ -25,50 +26,223 @@ void SinglePlayerGame::scanSpaces() {
+                }
+            }
+        }
+        if (candidatePieceIndex.size() >= 2){
+            priorityList.push_back(candidatePieceIndex);
+        }
+    }
+}
+
+
+// Scans the board for priority population
+void SinglePlayerGame::priorityScan() {
+    unsigned int i;
+    unsigned int j;
+    unsigned int blackCounter;
+    unsigned int whiteCounter;
+    int emptyIndex;
+    
+    // Clears the vectors for repopulation
+    possibleMill.clear();
+    possibleBlock.clear();
+    priorityList.clear();
+    
+    // Scans the board to find possible mills and blocks
+    for (i = 0; i < millList.size(); i++){
+        blackCounter = 0;
+        whiteCounter = 0;
+        // Flag for existence of an empty space in the line
+        emptyIndex = -1;
+        for (j = 0; j < 3; j++){
+            if (spaceList[millList[i][j]]->hasWhitePiece()){
+                whiteCounter++;
+            } else if (spaceList[millList[i][j]]->hasBlackPiece()){
+                blackCounter++;
+            } else{
+              // If an empty space is found
+              emptyIndex = millList[i][j];
+            }
+            // If there's no empty spaces, the line is ignored
+            if (whiteCounter == 2 && emptyIndex != -1){
+                (computerColorWhite) ? possibleMill.push_back(emptyIndex) : possibleBlock.push_back(emptyIndex);
+            }
+            if (blackCounter == 2 && emptyIndex != -1) {
+                (!computerColorWhite) ? possibleMill.push_back(emptyIndex) : possibleBlock.push_back(emptyIndex);
+            }
+        }
+    }
+}
+
+
 //Chooses an available space and moves in first phase
 void SinglePlayerGame::computerPhaseOneMove() {
-    scanSpaces();
-    int randIndex = rand() % availableSpaces.size();
-    spaceList[availableSpaces[randIndex]]->computerClickSpace();
+    priorityScan();
+    /* Going to want to prioritize mills over blocks.
+     * If no mill, find blocks. If no blocks or mills,
+     * find a random space*/
+    if (!possibleMill.empty()){
+        spaceList[possibleMill[0]]->computerClickSpace();
+    } else if (!possibleBlock.empty()){
+        spaceList[possibleBlock[0]]->computerClickSpace();
+    } else {
+        scanSpaces();
+        int randIndex = rand() % availableSpaces.size();
+        spaceList[availableSpaces[randIndex]]->computerClickSpace();
+    }
 }
 
+
 //Chooses a piece and moves to an open space
 void SinglePlayerGame::computerPhaseTwoMove() {
     unsigned int i;
     bool validMove = false;
     Piece *chosenPiece;
-    //Loops until a piece with an open adjacent space is selected
-    while (!validMove) {
-        //Choosing a piece
-        int randIndex = rand() % availableSelect.size();
-        int pieceIndex = availableSelect[randIndex];
+    int moveSpaceIndex;
+
+    priorityScanPhaseTwo();
+    if (!priorityList.empty()){
+        validMove = true;
+        int pieceIndex = availableSelect[priorityList[0][1]];
         if (computerColorWhite) {
```

Comment:
Later is there any way we could randomize this instead of picking the first available choice in priorityList?